### ISSUE TITLE
nixos-install failing with nix-1.12 due to missing Perl

### ISSUE THREAD

**Issue by Infinisil** [NixOS/nixpkgs#28595]
## Issue description

With nix 1.12, `nixos-install` fails at [this line](https://github.com/NixOS/nixpkgs/blob/ee1f1216c6bd7394594b3ed69f5029ba2ebcf648/nixos/modules/installer/tools/nixos-install.sh#L106) due to Perl missing (as the TODO above states). The error this gives is (for searchability)

```
Can't open perl script "use Nix::Config; Nix::Config::readConfig; print $Nix::Config::config{"binary-caches"};": No such file or directory
```

### Steps to reproduce

- Set `nix.package = pkgs.nixUnstable;` and rebuild
- Call `nixos-install`

## Technical details

* System: 17.09.git.96457d2 (Hummingbird)
* Nix version: nix (Nix) 1.12pre5511_c94f3d55
* Nixpkgs version: 17.09.git.96457d26dd
* Sandboxing enabled: Yes

**Comment by fpletz** [comment]
Not a blocker for 17.09 because Nix 1.12 will not the default yet.

**Comment by vcunat** [comment]
Should be fixed by e88f28965a7d.  (Installer tests now work with Nix 2.0.)
